Tests/UnitTests/FoundationExtensions/DecoderExtensionTests.swift

```diff
@@ -108,6 +108,16 @@ class DecoderExtensionsLossyCollectionTests: XCTestCase {
         expect(decodedData) == data
     }
 
+    func testDictionaryKeysAreSnakeCase() throws {
+        let data = Data(list: [],
+                        map1: ["snake_case": .init(string: "test1")],
+                        map2: [:])
+        let decodedData = try data.encodeAndDecode()
+
+        expect(decodedData) == data
+        expect(Set(decodedData.map1.keys)) == ["snake_case"]
```

Comment:
This was becoming `snakeCase`, which was breaking the deserialization of [these tests](https://github.com/RevenueCat/purchases-ios/blob/4.3.0/Tests/UnitTests/Purchasing/CustomerInfoTests.swift#L58) for example.